### PR TITLE
Tiberian Dawn Lua code cleanup

### PR DESCRIPTION
**Pull request by Mailaender**
Split from https://github.com/OpenRA/OpenRA/pull/20265.

### CODE REVIEW

**Review by abcdefg30 — DISMISSED** [review]

File: mods/cnc/maps/nod08b/nod08b-AI.lua [hunk 1]
```diff
@@ -158,30 +158,30 @@ Trigger.OnAllKilledOrCaptured(GDIBase, function()
 	Utils.Do(GDI.GetGroundAttackers(), IdleHunt)
 end)
 
-Trigger.OnKilled(GDIProc, function(building)
+Trigger.OnKilled(GDIProc, function()
```
Comment (anvilvapre) [inline mods/cnc/maps/nod08b/nod08b-AI.lua:161]: LUA is flexible in the type of function that you pass?

I will also not lead to a memory leak of the building argument. I guess it's reference counted and it won't.

Comment (abcdefg30) [reply mods/cnc/maps/nod08b/nod08b-AI.lua:161]: Yes, you don't need to specify the parameters if you don't use them.

Comment (Mailaender) [reply mods/cnc/maps/nod08b/nod08b-AI.lua:161]: The IDE showed it as greyed out, but didn't issue a warning.

File: mods/cnc/maps/gdi04b/gdi04b.lua [hunk 2]
```diff
@@ -144,24 +144,24 @@ WorldLoaded = function()
...
+			AutoTriggered = true
 			Trigger.RemoveFootprintTrigger(id)
 
 			Build(AutoUnits, true, function(actor)
 				Trigger.OnKilled(actor, NodUnitKilled)
 				IdleHunt(actor)
 			end)
 
-			Trigger.AfterDelay(DateTime.Seconds(4), function()
+			Trigger.AfterDelay(DateTime.Seconds(4), function(tank)
```
Comment (RoosterDragon) [inline mods/cnc/maps/gdi04b/gdi04b.lua]: `Trigger.AfterDelay` doesn't provide any callback args. Should `tank` be defined elsewhere?

File: mods/cnc/maps/gdi04c/gdi04c.lua [hunk 3]
```diff
@@ -41,7 +40,6 @@ end
 
 TownAttackAction = function(actor)
 	Trigger.OnIdle(actor, TownAttackersIdleAction)
-	FollowWaypoints(actor, TownAttackWpts)
```
Comment (RoosterDragon) [inline mods/cnc/maps/gdi04c/gdi04c.lua:44]: Why remove this?

Comment (Mailaender) [reply mods/cnc/maps/gdi04c/gdi04c.lua:44]: `waypoint1` and `waypoint2` are local `nil` values. This is dead code.

**Review by RoosterDragon — APPROVED** [review]

**Review by PunkPun — APPROVED** [review]

**Comment by PunkPun** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/c202953afddca5cf2c28dacd2a24dd30b03390cf...de00e0510cd36b7048497031b6663a9efd6c7b6f)

**Pull request merged**